docs/essentials/widget_language.md

```diff
@@ -181,7 +182,7 @@ DynWidget {
 }
 ``` 
 
-This `DynWidget` generates an optional widget as detected by if the `counter` is greater than zero. One thing that could be improved is to the `greet` not always regenerate after the `counter` modified. Indeed the `greet` widget can regenerate only if the result `*counter > 0` changed. Let's go further:
+This `DynWidget` generates an optional widget when it detects that the `counter` is greater than zero. One thing that could be improved is to not always regenerate `greet` after the `counter` changes. Indeed the `greet` widget should regenerate only after the result `*counter > 0` changed the first time. Let's go further:
```

Comment:
> ... regenerate only after the result `*counter > 0` changed **the first time**

There is some ambiguity in my original text, so you can't append "the first time" here, the dynamic widget will not regerenerate only once.  What I mean is:

Indeed, the `greet` widget can only be regenerated if the result of the whole `*counter > 0` has changed, not just the `counter` change.